tests/integration/integration-test.py

```diff
@@ -46,47 +48,38 @@ def test_get_pets_with_filter(self, endpoint='/pets'):
 
     # Test case: GET /pets with pagination parameters
     def test_get_pets_pagination(self, endpoint='/pets'):
-        testing_valid_paginations = [
+        testing_paginations = [
             {'number': 1, 'size': 25, 'expected_status_code': 200},
             {'number': 1, 'size': None, 'expected_status_code': 200},
             {'number': None, 'size': 25, 'expected_status_code': 200},
-            {'number': 999, 'size': 1, 'expected_status_code': 200}
-        ]
-
-        testing_invalid_paginations = [
+            {'number': 999, 'size': 1, 'expected_status_code': 200},
             {'number': -1, 'size': 25, 'expected_status_code': 400},
             {'number': 1, 'size': -1, 'expected_status_code': 400},
             {'number': 1, 'size': 501, 'expected_status_code': 400}
         ]
-
-        for pagination in testing_valid_paginations:
+        nullable_fields = ['owner']
+        for pagination in testing_paginations:
             params = {f'page[{k}]': pagination[k] for k in ['number', 'size']}
             expected_status_code = pagination['expected_status_code']
             response = utils.test_endpoint(self, endpoint,
-                                           resource='PetResource',
+                                           resource='PetResource' if
+                                           expected_status_code is 200
+                                           else 'Error',
```

Comment:
Not sure, but I guess directly using ternary operation in the argument assignment looks a bit odd to me. I'd probably use a variable to store it as you did for the `expected_status_code = pagination['expected_status_code']` before passing it into the `test_endpoint` function to avoid the confusion.